modules/data.land/R/IC_BADM_Utilities.R

```diff
@@ -0,0 +1,296 @@
+entries <- entries[-which(ind),]
+
+
+
+  return(entries)
+}
+
+
+#' netcdf.writer.BADAM
+#'
+#' @param lat numeric latitude
+#' @param long numeric longitude
+#' @param siteid site id as a string
+#' @param outdir outputdir which you want to store the IC netcdf file
+#'
+#' @return a string of the address where the IC file is stored
+#' @export
+#'
+netcdf.writer.BADAM <- function(lat, long, siteid, outdir ){
+  #--
+  input <- list()
+  dims <- list(lat = lat ,
+               lon = long,
+               time = 1)
+
+  #Reading in the BADM data
+  entries <- Read.IC.info.BADM (lat, long)
+  
+  PWI <- entries$PlantWIni[!is.na(entries$PlantWIni)]
+  LIn <- entries$LitterIni[!is.na(entries$LitterIni)]
+  SIn <- entries$SoilIni[!is.na(entries$SoilIni)]
+  
+  
+  variables <- list(
+    SoilMoistFrac = 1,
+    SWE = 0
+  )
+  
+  if (length(PWI)>0 ) variables <- c(variables, wood_carbon_content = sample(PWI, 1, replace = T))
+  if (length(LIn)>0 ) variables <- c(variables, litter_carbon_content = sample(LIn, 1, replace = T))
+  if (length(SIn)>0 ) variables <- c(variables, soil_organic_carbon_content = sample(SIn, 1, replace = T))
+  
+  input$dims <- dims
+  input$vals <- variables
+  
+  return(pool_ic_list2netcdf(
+    input = input,
+    outdir = outdir,
+    siteid = siteid
+  )$file)
+}
+
+
+#' IC_Maker
+#'
+#' @param settings PEcAn xml settings
+#' @param ens.n number of ensemble memebers needs to be genrated
+#' @param outdir output directory 
+#' @description This function generates a series of ensemble members for initial condition using the BADM database.
+#' @return xml setting
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' settings <- PEcAn.settings::read.settings("pecan.SDA.10sites.xml")
+#' 
+#'  suppressMessages({
+#'   setting.with.ic <- PEcAn.settings::papply(settings,
+#'                           IC_Maker,
+#'                           ens.n=10,
+#'                           outdir="/fs/data3/hamzed/MultiSite_Project/IC/ICFiles")
+#' })
+#' } 
+IC_Maker <-function(settings, ens.n=5, outdir) {
+  
+
+  ini.pool <- 1:ens.n %>%
+    purrr::map(
+      ~ netcdf.writer.BADAM(
+        settings$run$site$lat %>% as.numeric(),
+        settings$run$site$lon %>% as.numeric(),
+        paste0( settings$run$site$id, .x) %>% as.numeric(),
+        outdir
+      )
+    ) %>%
+    setNames(rep('path', ens.n)) %>%
+    list %>%
+    setNames('poolinitcond')
+  
+  settings$run$inputs <-
+    c(settings$run$inputs, ini.pool)
+  
+  return(settings)
+}
+
+
+#' L1_L2_finder
+#'
+#' @param Lat numeric latitude
+#' @param Lon numeric longitude
+#' @description This function is designed to find the level1 and level2 code ecoregions for a given lat and long. 
+#' You can learn more about ecoregions here: \link{https://www.epa.gov/eco-research/ecoregions}.
+
+#'
+#' @return a dataframe with codes corresponding to level1 and level2 codes as two columns
+#' @export
+#'
+L1_L2_finder <- function(Lat, Lon){
```

Comment:
Function  naming:  If you name this something a bit more generic (e.g. EPA_ecoregion_finder) then it will be easier to generalize this in the future (e.g. if you want it to later return L3 or L4 also) and easier for others to understand what it does and remember how to call it.